### DISCUSSION TITLE
LDAP Users get removed from Group on login with Unauthorized to view the page

### DISCUSSION THREAD

**Discussion by s0p4L1n3** [requarks/wiki#6215] [Error / Bug Report] [answered]
Hello,

I just installed a fresh WikiJS on Oracle Linux 8 VM. Configured Active Directory, created 1 group Admin-AD.
Everything works fine except the fact on the first login or if I logout and login again, I get Not Authorized page and in Incognito mode if I check with local admin user I can see my Active Directory user has been automatically removed from the Administrator local group and the second one I created.

The steps to fix this is:

- in incognito mode, log in with the local admin, assign my LDAP user again to both groups. If I close the browser completely and open again the URL, I have access to everything with the proper permissions.

Is this bug known to someone or it can be some settings misconfigured on the page permissions ?

In the group Admin-AD, the Redirect on Login is /fr/home
The group has all permissions
The page rules contain all selected permissions with **Paths start with** -->  /

I enabled LDAP Debug to find some error in the logs, but no files are generated, in the console nor in a file.

**Comment by q-wertz** [comment, marked as answer]
Not 100% sure if this also applies to the Administrator local group, but:

Do you have group mapping enabled? If so, your LDAP users are removed from groups they are not part of in your ActiveDirectory tree (see also https://github.com/requarks/wiki/commit/8fa771c4ce6b667d5c2b1de6b343977af4351644#diff-e62bba4f65b928357a9c4a9fbc4f794c486677f609f1ea3945bb56937a7f937eR50) 

You would need to create a group in your LDAP with the exact same name and adjust the WikiJS configuration to make sure the groups are found.

**Reply by s0p4L1n3** [reply]
Thank you, it was it. I though the groups mapping was usefull to allow access to specific LDAP groups if we wanted to. 
I did not know/did not read the note about the removed group info (my fault)

**Reply by q-wertz** [reply]
You can also allow/deny access to specific groups. 
1. You can adapt the LDAP group filter to exclude specific AD groups or only include certain
2. I think you can also just neglect adding groups in Wiki that are you do not want synchronized. No groups are currently added in Wiki by the authentication 

Just another note regarding the admin group: I would create separate local admin users for Wiki. (I think when you loose connection to the AD server you cannot login using your AD accounts and are locked out)

**Reply by s0p4L1n3** [reply]
Thanks for the help !

**Reply by icsinfo** [reply]
Also to help, you can't create a separate admin group that has the manage:system permission from the front end, you have to use a graph ql query. This worked for us. Just create a new group and using its group id and name in the groups list, run the following graphql query from the playground.

```graphql
mutation {
  groups {
    update(
      id: your group id number
      name: "Your Group Name"
      permissions: ["manage:system"]
      redirectOnLogin: "/"
      pageRules: {
        id: "default"
        deny: false
        path: ""
        locales: []
        match: START
        roles: []
      }
    ) {
      responseResult {
        succeeded
        slug
        message
      }
    }
  }
}
```
Hopefully this will help anyone that doesn't want to sync users between the built in Administrators group in wiki.js and AD.
